tests/libgit2/online/shallow.c

```diff
@@ -164,3 +164,102 @@ void test_online_shallow__unshallow(void)
 	git_revwalk_free(walk);
 	git_repository_free(repo);
 }
+
+void test_online_shallow__deepen_six(void)
+{
+	git_str path = GIT_STR_INIT;
+	git_repository *repo;
+	git_revwalk *walk;
+	git_clone_options clone_opts = GIT_CLONE_OPTIONS_INIT;
+	git_fetch_options fetch_opts = GIT_FETCH_OPTIONS_INIT;
+	git_remote *origin = NULL;
+	git_oid oid;
+	git_oid *roots;
+	size_t roots_len;
+	size_t num_commits = 0;
+	int error = 0;
+
+	clone_opts.fetch_opts.depth = 5;
+	clone_opts.remote_cb = remote_single_branch;
+
+	git_str_joinpath(&path, clar_sandbox_path(), "deepen_6");
+	cl_git_pass(git_clone(&repo, "https://github.com/libgit2/TestGitRepository", git_str_cstr(&path), &clone_opts));
+	cl_assert_equal_b(true, git_repository_is_shallow(repo));
+
+	fetch_opts.depth = 6;
+	cl_git_pass(git_remote_lookup(&origin, repo, "origin"));
+	cl_git_pass(git_remote_fetch(origin, NULL, &fetch_opts, NULL));
+	cl_assert_equal_b(true, git_repository_is_shallow(repo));
+
+	cl_git_pass(git_repository__shallow_roots(&roots, &roots_len, repo));
+	cl_assert_equal_i(4, roots_len);
+	cl_assert_equal_s("58be4659bb571194ed4562d04b359d26216f526e", git_oid_tostr_s(&roots[0]));
+	cl_assert_equal_s("d31f5a60d406e831d056b8ac2538d515100c2df2", git_oid_tostr_s(&roots[1]));
+	cl_assert_equal_s("6462e7d8024396b14d7651e2ec11e2bbf07a05c4", git_oid_tostr_s(&roots[2]));
+	cl_assert_equal_s("2c349335b7f797072cf729c4f3bb0914ecb6dec9", git_oid_tostr_s(&roots[3]));
+
+	git_revwalk_new(&walk, repo);
+	git_revwalk_push_head(walk);
+
+	while ((error = git_revwalk_next(&oid, walk)) == GIT_OK) {
+		num_commits++;
+	}
+
+	cl_assert_equal_i(num_commits, 17);
+	cl_assert_equal_i(error, GIT_ITEROVER);
+
+	git__free(roots);
+	git_remote_free(origin);
+	git_str_dispose(&path);
+	git_revwalk_free(walk);
+	git_repository_free(repo);
+}
+
+void test_online_shallow__shorten_four(void)
```

Comment:
```suggestion
void test_online_shallow__shorten_five(void)
```

It looks like this actually deepens five not four?

Reply:
Both `test_online_shallow__deepen_six()` and `test_online_shallow__shorten_four()` start by cloning the test repository with a depth of 5 (`clone_opts.fetch_opts.depth = 5;`).

`test_online_shallow__deepen_six()` then fetches from the repository again, but with a depth of 6 (`fetch_opts.depth = 6;`), to demonstrate that libgit2 is able to make a shallow history *deeper*.

`test_online_shallow__shorten_four()`, which you called out in your comment, performs the post-cloning fetch with a depth of 4 (`fetch_opts.depth = 4;`), to demonstrate that libgit2 is also able to make a shallow history *shorter*.

Does this help?

Reply:
Oops. 😓  Yes, sorry, I misread that.